### PR TITLE
[12.0][MIG] rma_analytic

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
The analytic propagation to invoices is missing. It is also missing in v10. I will that before merging

File: rma_analytic/tests/test_rma_analytic.py [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
-        res.update({'analytic_account_id': analytic_1.id})
+        analytic_1 = cls.env["account.analytic.account"].create(
+            {"name": "Test account #1"}
+        )
+        res.update({"analytic_account_id": analytic_1.id})
         return res
 
-    def test_analytic(self):
-        for line in self.rma_ana_id.rma_line_ids:
+    def test_analytic(cls):
```
Comment (MiquelRForgeFlow) [inline rma_analytic/tests/test_rma_analytic.py]: here use `self` instead

File: rma_analytic/wizards/rma_make_picking.py [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
 
 from odoo import models, api
 
 
 class RmaMakePicking(models.TransientModel):
-    _name = 'rma_make_picking.wizard'
-    _description = 'Wizard to create pickings from rma lines'
+    _inherit = "rma_make_picking.wizard"
+    _description = "Wizard to create pickings from rma lines"
```
Comment (MiquelRForgeFlow) [inline rma_analytic/wizards/rma_make_picking.py]: no need description here

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

File: rma_analytic/tests/test_rma_analytic.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply rma_analytic/tests/test_rma_analytic.py]: classmethod should work, shouldn't it?

Comment (MiquelRForgeFlow) [reply rma_analytic/tests/test_rma_analytic.py]: in the testing methods named `test_something` no, they always are with self and without classmethod

Comment (MiquelRForgeFlow) [reply rma_analytic/tests/test_rma_analytic.py]: i_dont_know_what_am_i_doing.jpeg

Comment (MiquelRForgeFlow) [reply rma_analytic/tests/test_rma_analytic.py]: I think you could to the same in tests of other modules like rma

Comment (AaronHForgeFlow) [reply rma_analytic/tests/test_rma_analytic.py]: Not priority IMHO, I think it is better for 13.0

File: rma/models/rma_order_line.py [hunk 3]
```diff
@@ -622,3 +622,9 @@ def action_view_rma_lines(self):
             result['views'] = [(res and res.id or False, 'form')]
             result['res_id'] = rma_lines[0]
         return result
+
+    @api.constrains("partner_id", "rma_id")
+    def _check_partner_id(self):
+        if self.rma_id and self.partner_id != self.rma_id.partner_id:
+            raise UserError(_(
```
Comment (MiquelRForgeFlow) [inline rma/models/rma_order_line.py]:
```suggestion
            raise ValidationError(_(
```

Comment (AaronHForgeFlow) [reply rma/models/rma_order_line.py]: This is another module. Solved here: https://github.com/Eficent/stock-rma/pull/73

**Comment by AaronHForgeFlow** [comment]
ping @mreficent good with rma_analytic?

File: rma_analytic/README.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
+.. image:: https://img.shields.io/badge/license-LGPLv3-blue.svg
```
Comment (MiquelRForgeFlow) [inline rma_analytic/README.rst]:
```suggestion
.. image:: https://img.shields.io/badge/license-LGPL--3-blue.png
```

File: rma_analytic/README.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+   :target: https://www.gnu.org/licenses/lgpl.html
```
Comment (MiquelRForgeFlow) [inline rma_analytic/README.rst]:
```suggestion
   :target: https://www.gnu.org/licenses/lgpl
```

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**